knotx-core/knotx-engine-view/src/main/java/com/cognifide/knotx/engine/view/impl/TemplateEngine.java

```diff
@@ -47,13 +49,17 @@ public TemplateEngine(EventBus eventBus, ViewEngineConfiguration configuration)
   }
 
   public Observable<String> process(KnotContext knotContext) {
-    return extractFragments(knotContext.template())
+    return Observable.just(knotContext.fragments())
+        .filter(Optional::isPresent)
+        .map(Optional::get)
+        .flatMap(Observable::from)
```

Comment:
observables with optionals hard to read here. 
instead do it simpler (optional have functional interface also):
```java
knotxContext.fragments().map(fragments -> Observable.from(fragments).do.rest.stuff  ).orElse(Observable.just(StringUtils.EMPTY)
```